test/gff_fasta_plant_upload_test.py

```diff
@@ -0,0 +1,118 @@
+from GenomeFileUtil.GenomeFileUtilImpl import GenomeFileUtil
+from GenomeFileUtil.GenomeFileUtilServer import MethodContext
+
+from DataFileUtil.DataFileUtilClient import DataFileUtil
+
+from GenomeFileUtil.JsonIOHelper import (download_genome_to_json_files, 
+                                         compare_genome_json_files)
+
+
+class GenomeFileUtilTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        token = environ.get('KB_AUTH_TOKEN', None)
+        # WARNING: don't call any logging methods on the context object,
+        # it'll result in a NoneType error
+        cls.ctx = MethodContext(None)
+        cls.ctx.update({'token': token,
+                        'provenance': [
+                            {'service': 'GenomeFileUtil',
+                             'method': 'please_never_use_it_in_production',
+                             'method_params': []
+                             }],
+                        'authenticated': 1})
+        config_file = environ.get('KB_DEPLOYMENT_CONFIG', None)
+        cls.cfg = {}
+        config = ConfigParser()
+        config.read(config_file)
+        for nameval in config.items('GenomeFileUtil'):
+            cls.cfg[nameval[0]] = nameval[1]
+        cls.wsURL = cls.cfg['workspace-url']
+        cls.wsClient = workspaceService(cls.wsURL, token=token)
+        cls.serviceImpl = GenomeFileUtil(cls.cfg)
+
+    @classmethod
+    def tearDownClass(cls):
+        if hasattr(cls, 'wsName'):
+            cls.wsClient.delete_workspace({'workspace': cls.wsName})
+            print('Test workspace was deleted')
+
+    def getWsClient(self):
+        return self.__class__.wsClient
+
+    def getWsName(self):
+        if hasattr(self.__class__, 'wsName'):
+            return self.__class__.wsName
+        suffix = int(time.time() * 1000)
+        wsName = "test_GenomeFileUtil_" + str(suffix)
+        ret = self.getWsClient().create_workspace({'workspace': wsName})
+        self.__class__.wsName = wsName
+        return wsName
+
+    def getImpl(self):
+        return self.__class__.serviceImpl
+
+    def getContext(self):
+        return self.__class__.ctx
+
+    def test_simple_upload(self):
+        # fetch the test files and set things up
+        genomeFileUtil = self.getImpl()
+        fasta_path = "data/Test_Plant/Test_v1.0.fa.gz"
+        gff_path = "data/Test_Plant/Test_v1.0.gene.gff3.gz"
+
+        ### Test for a Local Function Call
+        print('attempting upload via local function directly')
+        ws_obj_name = 'MyGenome'
+        result = genomeFileUtil.fasta_gff_to_genome(self.getContext(), 
+            {
+                'fasta_file' : { 'path': fasta_path },
+                'gff_file' : { 'path' : gff_path },
+                'workspace_name':self.getWsName(),
+                'genome_name':ws_obj_name
+            })[0]
+        pprint(result)
+        self.assertIsNotNone(result['genome_ref'])
```

Comment:
This test coverage is really light, and there are no checks to make sure the genome saved is what you expect.  At the least, we should check a few metadata fields, maybe the feature count, possibly that the correct Assembly is referenced.

If we're planning to support FTP uploads, we should also add a test that pulls the input via ftp.  Are there public ftp sites that are fairly stable we can use?  For genomes, we use ncbi...